sdk/platform/http_client/curl/src/az_curl.c

```diff
@@ -519,18 +559,20 @@ static AZ_NODISCARD az_result _az_http_client_curl_send_request_impl_process(
   {
     result = _az_http_client_curl_send_get_request(p_curl);
   }
-  else if (az_span_is_content_equal(p_request->_internal.method, az_http_method_post()))
-  {
-    result = _az_http_client_curl_send_post_request(p_curl, p_request);
-  }
   else if (az_span_is_content_equal(p_request->_internal.method, az_http_method_delete()))
   {
     result = _az_http_client_curl_send_delete_request(p_curl, p_request);
   }
+  else if (az_span_is_content_equal(p_request->_internal.method, az_http_method_post()))
```

Comment:
Why was this else if branch moved to  a different order? Was this intentional and does the behavior change because of it?

Reply:
intentional to have PUT and POST together.   Nothing really important